### ISSUE TITLE
Feature request: Customizing the query

### ISSUE THREAD

**Issue by mikeconley12** [feathersjs-ecosystem/feathers-objection#123]
It would be great to have an ability to customize the query. See [the Knex service adapter docs](https://github.com/feathersjs-ecosystem/feathers-knex#customizing-the-query) for details

**Comment by dekelev** [comment]
You do have the ability. Use modifiers methods in your model and run them
on queries with $modify.

בתאריך שבת, 17 באוק׳ 2020, 16:18, מאת mikeconley12 ‏<
notifications@github.com>:

**Comment by mikeconley12** [comment]
Thanks for your answer! I will check the modifiers in a few days and close this issue or explain why it doesn't work.

**Comment by mikeconley12** [comment]
Hi @dekelev ! I have a question.
How can I specify multiple modifiers with different arguments?

I need an equivalent to this:

```js
const women = await Person.query()
  .modify('filterAge', 18)
  .modify('filterGender', 'female');
```

**Comment by dekelev** [comment]
You can add 2 modifiers to the model and when calling find on the service,
add params.query.$modify=modifier1_fn_name,modifier2_fn_name

Check the docs for examples.

בתאריך יום א׳, 18 באוק׳ 2020, 19:07, מאת mikeconley12 ‏<
notifications@github.com>:

**Comment by dekelev** [comment]
Passing arguments to modifiers is supported too, but I think you'll need to
use array for that.

בתאריך יום א׳, 18 באוק׳ 2020, 20:01, מאת Dekel ‏<dekelev@gmail.com>:

**Comment by mikeconley12** [comment]
yes, I checked the docs. But I need different arguments for each modifier. Is it possible?

```js
params.query.$modify = { "modifier1_fn_name": ["args1"], "modifier2_fn_name": ["args2"] };
```

**Comment by dekelev** [comment]
Yes, it's possible

בתאריך יום א׳, 18 באוק׳ 2020, 20:15, מאת mikeconley12 ‏<
notifications@github.com>:

**Comment by dekelev** [comment]
Check the test file for usage examples.

בתאריך יום א׳, 18 באוק׳ 2020, 20:44, מאת Dekel ‏<dekelev@gmail.com>:

**Comment by mikeconley12** [comment]
Sorry, but I already checked [this test file](https://github.com/feathersjs-ecosystem/feathers-objection/blob/master/test/index.test.js) and I didn't find a solution.

For example, I see [here](https://github.com/feathersjs-ecosystem/feathers-objection/blob/master/test/index.test.js#L2105) multiple modifiers, but all the modifiers have the same arguments.

**Comment by dekelev** [comment]
https://github.com/feathersjs-ecosystem/feathers-objection/blob/0f4a36702a4b134933650afd81d2bafaaa5812f5/test/index.test.js#L2097

Each item in the array is an array that its first item is the modifier name
and rest of items are args for that modifier.

בתאריך יום א׳, 18 באוק׳ 2020, 21:03, מאת mikeconley12 ‏<
notifications@github.com>:

**Comment by mikeconley12** [comment]
> and rest of items are args for that modifier

No, in your example, `large` is a modifier https://github.com/feathersjs-ecosystem/feathers-objection/blob/dbd6e447116b08ede16133d14d7fc9ce6dff18ce/test/company.js#L53

Also see Objection.js [documentation](https://vincit.github.io/objection.js/api/query-builder/other-methods.html#modify). It says that the first argument is a modifier, and the rest are arguments passed to the modify function

**Comment by dekelev** [comment]
Right, so I'll add support for JSON object as value for `$modify` and we'll be able to use it as you suggested and without a breaking-change:

`{ "modifier1_fn_name": ["args1"], "modifier2_fn_name": ["args2"] }`

It's also much more readable in my opinion, so it would also be my default value type for `$modify`.
I'll add examples to the docs once it's ready, but basically it's a low effort task, so I hope to release it later today.

**Comment by mikeconley12** [comment]
It's awesome! Thank you! But will it be a JSON string? Or just a regular JS object?

**Comment by dekelev** [comment]
It will also support stringified JSON object.

בתאריך יום ב׳, 19 באוק׳ 2020, 13:08, מאת mikeconley12 ‏<
notifications@github.com>:

**Comment by dekelev** [comment]
Released with v6.1.0

I've updated the [$modify docs](https://github.com/feathersjs-ecosystem/feathers-objection#query-operators) as well. enjoy!

**Comment by mikeconley12** [comment]
Great, thanks for your help!
